Explain the need to use underscores in config yaml (#2204)
kedro-org/kedro: docs/source/kedro_project_setup/configuration.md

  env: env1
```

The syntax for the options is different when you're using the CLI compared to the configuration file. In the CLI you use dashes, for example for `kedro run --from-nodes ...`, but you have to use an underscore in the configuration file:

```yaml
run:
  from_nodes: ...
```

> This is because further down the process the configuration file gets parsed by [Click](https://click.palletsprojects.com/en/8.1.x/), a python package to handle command line interfaces. Click passes the options defined in the configuration file directly to a python function and so the option names need to match the argument names in that function.
> Variable names and argument in Python can only contain alpha-numeric characters and underscores, so it's not possible to have a dash in the option names when using the configuration file.

Reviewer: ```suggestion
This is because the configuration file gets parsed by [Click](https://click.palletsprojects.com/en/8.1.x/), a Python package to handle command line interfaces. Click passes the options defined in the configuration file to a Python function. The option names need to match the argument names in that function.

Variable names and arguments in Python may only contain alpha-numeric characters and underscores, so it's not possible to have a dash in the option names when using the configuration file.
```

Reviewer: Just a few minor tweaks. I think we need to capitalise Python on every usage.
